en/cpp/api_changes.md

```diff
@@ -0,0 +1,121 @@
+# API Changes
+
+This page tracks changes between versions.
+
+It covers both breaking (incompatible) and non-breaking changes.
```

Comment:
@julianoes I removed the note about not wanting to break - that is implied in all software IMO.